src/controller/app_routes.rs

```diff
@@ -177,49 +177,69 @@ impl AppRoutes {
         app = Self::add_powered_by_header(app, &ctx.config.server);
 
         if let Some(catch_panic) = &ctx.config.server.middlewares.catch_panic {
-            if catch_panic.enable {
+            if catch_panic.is_enabled() {
                 app = Self::add_catch_panic(app);
             }
         }
 
         if let Some(compression) = &ctx.config.server.middlewares.compression {
-            if compression.enable {
+            if compression.is_enabled() {
                 app = Self::add_compression_middleware(app);
             }
         }
 
-        if let Some(limit) = &ctx.config.server.middlewares.limit_payload {
-            if limit.enable {
-                app = Self::add_limit_payload_middleware(app, limit)?;
-            }
+        if let Some(limit) = ctx
+            .config
+            .server
+            .middlewares
+            .limit_payload
+            .clone()
```

Comment:
any way to shorten this chain? I'm not sure myself just wondering, because if it ends up in a ref eventually, maybe we can supply a ref directly

Reply:
Jup. A little at least. Its a bit annoying that it is `Option<Enable<...>>` which means I have to poke through those containers a little.